app_tests/integration_tests/llm/model_management.py

```diff
@@ -275,4 +290,37 @@ def process_model(self, config: ModelConfig) -> ModelArtifacts:
             mlir_path=mlir_path,
             vmfb_path=vmfb_path,
             config_path=config_path,
+            model_config=config,
         )
+
+
+TEST_MODELS = {
+    "open_llama_3b": ModelConfig(
+        source=ModelSource.HUGGINGFACE,
+        repo_id="SlyEcho/open_llama_3b_v2_gguf",
+        model_file="open-llama-3b-v2-f16.gguf",
+        tokenizer_id="openlm-research/open_llama_3b_v2",
+        batch_sizes=(1, 4),
+        device_settings=device_settings.get_device_based_on_env_variable(),
```

Comment:
Not sure if this is the best way to set up the device, but doing this because ModelConfig can be used as a cache key for the stored model artifacts, and the device settings matter for caching compiled artifacts.

Reply:
Using a get_device_settings_by_name function instead.